<p>
    All strings should be enclosed in single quotes, unless they contain variables or the following characters:
    <code>"</code> (double quotes), <code>'</code> (single quote), <code>\n</code>, <code>\r</code>, <code>\t</code>.
    Note that if the string only contains a variable, quotes should be removed. An exception though on resource title
    that should always be quoted even if they only contain a variable.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
abc { 'foo':
  text1 => "abc",                # Noncompliant, no variables, no special characters => single quotes should be used
  text2 => "Dollar sign is $",   # Noncompliant, no variables, no special characters => single quotes should be used
  text3 => 'Don\'t do it',       # Noncompliant, double quotes should be used and the single quote should be unescaped
  text4 => "$var",               # Noncompliant, variable only => quotes should be removed
  text5 => "${var}",             # Noncompliant, variable only => quotes should be removed
}
</pre>
<h2>Compliant Solution</h2>
<pre>
abc { 'foo':
  text1 => 'var',
  text2 => 'Dollar sign is $',
  text3 => "Don't do it",
  text4 => $var,
  text5 => $var,
  text6 => "My variable is ${var}",
  text7 => "blabla\n blabla\n",
}

resource { "${title}":  # Compliant because it is a resource title
}
</pre>
<h2>See</h2>
<ul>
    <li><a href="https://docs.puppetlabs.com/guides/style_guide.html#quoting">See Puppet Labs Puppet Language Style
        Guide</a></li>
</ul>
