<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all variables match the following regular expression: <nobr><code>^(::)?([a-z][a-z0-9_]*::)*[a-z][a-z0-9_]*$</code></nobr></p>
<p>
  This naming convention is compliant with the more strict Puppet 4 naming convention.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
file { $myFile:    # Noncompliant
  ensure => file,
  mode   => $Mode,  # Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
file { $my_file:
  ensure => file,
  mode   => $mode,
}
</pre>
<h2>See</h2>
<ul>
  <li><a href="https://docs.puppetlabs.com/guides/style_guide.html#variable-format">See Puppet Labs Puppet Language Style Guide</a></li>
  <li><a href="https://docs.puppetlabs.com/puppet/latest/reference/experiments_future.html#check-names-of-variables-classes-functions-defined-types-etc">Updating 3.x Manifests for Puppet 4.x</a></li>
</ul>
